logbook/handlers.py

```diff
@@ -642,6 +649,66 @@ def ensure_stream_is_open(self):
             self._open()
 
 
+class GZIPCompressionHandler(FileHandler):
+    def __init__(self, filename, encoding=None, level=NOTSET,
+                 format_string=None, delay=False, filter=None, bubble=False,
+                 compression_window_size=4*1024**2, compression_quality=9):
+        self._compression_quality = compression_quality
+        self._compression_window_size = compression_window_size
+        self._num_unflushed_bytes = 0
+        FileHandler.__init__(self, filename, mode='wb', encoding=encoding, level=level,
+                             format_string=format_string, delay=delay, filter=filter, bubble=bubble)
+
+    def _open(self, mode=None):
+        if mode is None:
+            mode = self._mode
+        self.stream = gzip.open(self._filename, mode, compresslevel=self._compression_quality)
+
+    def write(self, item):
+        if isinstance(item, str):
+            item = item.encode(encoding=self.encoding)
+        self.ensure_stream_is_open()
+        self.stream.write(item)
+        self._num_unflushed_bytes += len(item)
+
+    def should_flush(self):
+        return self._num_unflushed_bytes >= self._compression_window_size
+
+    def flush(self):
+        super(GZIPCompressionHandler, self).flush()
+        self._num_unflushed_bytes = 0
+
+
+class BrotliCompressionHandler(FileHandler):
+    def __init__(self, filename, encoding=None, level=NOTSET,
+                 format_string=None, delay=False, filter=None, bubble=False,
+                 compression_window_size=4*1024**2, compression_quality=11):
+        FileHandler.__init__(self, filename, mode='wb', encoding=encoding, level=level,
+                             format_string=format_string, delay=delay, filter=filter, bubble=bubble)
+        max_window_size = int(math.log(compression_window_size, 2))
+        self._compressor = Compressor(quality=compression_quality, lgwin=max_window_size)
+
+    def _open(self, mode=None):
+        if mode is None:
+            mode = self._mode
+        self.stream = io.open(self._filename, mode)
+
+    def write(self, item):
+        if isinstance(item, str):
+            item = item.encode(encoding=self.encoding)
+        ret = self._compressor.process(item)
+        if ret:
+            self.ensure_stream_is_open()
+            self.stream.write(ret)
+
+    def close(self):
```

Comment:
flush is not implemented at all, meaning that flushing it externally will not flush the compressor...